src/textual/_cache.py

```diff
@@ -70,10 +86,9 @@ def grow(self, maxsize: int) -> None:
 
     def clear(self) -> None:
         """Clear the cache."""
-        with self._lock:
-            self._cache.clear()
-            self._full = False
-            self._head = []
+        self._cache.clear()
+        self._full = False
+        self._head = []
```

Comment:
Should it also clear hits and misses counts?

Reply:
The hits and misses are a debugging aid to see how well the cache is performing. You generally want to know about past performance even when it is cleared.